Symfony/CS/Fixer/Contrib/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Vincent Klaiber <hello@vinkla.com>
+ */
+class ShortEchoTagFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $newContent = preg_replace('/<\?=(\s|$)/', '<?php echo$1', $content, -1, $count);
+
+        if (!$count) {
+            return $content;
+        }
+
+        $tokens = Tokens::fromCode($newContent);
+        $tokensOldContent = '';
+        $tokensOldContentLength = 0;
+
+        foreach ($tokens as $token) {
+            if ($token->isGivenKind(T_OPEN_TAG)) {
+                $tokenContent = $token->getContent();
+
+                if ('<?=' === substr($content, $tokensOldContentLength, 3)) {
+                    $tokenContent = '<?php echo';
```

Comment:
this is wrong. You need to insert the proper tokens in the collection instead of putting ``<?php echo`` as the content of an T_OPEN_TAG. Otherwise you will corrupt the token collection, which will impact following fixers because of the tokenizer cache layer (and would be worse on the 2.x branch given the tokenizer runs only once)